keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/NotificationEvent.kt

```diff
@@ -1,13 +1,40 @@
 package com.netflix.spinnaker.keel.events
 
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.core.api.EnvironmentArtifactPin
+import com.netflix.spinnaker.keel.core.api.PinnedEnvironment
 import com.netflix.spinnaker.keel.notifications.Notification
 import com.netflix.spinnaker.keel.notifications.NotificationScope
 import com.netflix.spinnaker.keel.notifications.NotificationType
 
-data class NotificationEvent(
-  val scope: NotificationScope,
-  val ref: String,
-  val type: NotificationType,
-  val message: Notification
-)
+abstract class NotificationEvent{
+  abstract val scope: NotificationScope
+  open val ref: String? = null
```

Comment:
why the change to `ref: String?`?

Reply:
same question for message

Reply:
I've changed it to fit the new notifiactionEvent module, where you don't necessarily have the ref and message (we construct the message in the slack module, for instance).

Reply:
You could pass the value of the `type` enum from the sub-classes for `ref` here, that way you wouldn't need to make it nullable. Same with `message`. In short, I don't think those fields should be made nullable just to fit the new use case.

Reply:
@luispollo Yep I just pushed a change for it -- after talking with @emjburns , we've realized that we can have both repeated notifications (like unhealthy, which requires being persisted) and single notifications. I've decided to separate those two for now. I will definitely refactor it in the near future.